cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MRJobClient.java

```diff
@@ -82,6 +82,10 @@ public MRJobInfo getMRJobInfo(Id.Run runId) throws IOException, NotFoundExceptio
     JobStatus thisJob = findJobForRunId(jobs, runId);
 
     RunningJob runningJob = jobClient.getJob(thisJob.getJobID());
+    if (runningJob == null) {
+      LOG.warn("JobClient returned null for JobId: '{}'", thisJob.getJobID());
+      throw new NotFoundException(runId);
```

Comment:
ok, updated to throw IllegalStateException with appropriate message. Also changed caller to catch Exception and use backup, instead of only doing it upon IOE and NFE.